analyzers/src/SonarAnalyzer.CSharp/Rules/FieldShouldBeReadonly.cs

```diff
@@ -231,13 +229,17 @@ private void CollectFieldsFromPrefixUnaryExpressions()
 
                 private void CollectFieldsFromAssignments()
                 {
-                    var assignments = this.partialTypeDeclaration.Node.DescendantNodes()
+                    var assignments = partialTypeDeclaration.Node.DescendantNodes()
                         .OfType<AssignmentExpressionSyntax>()
-                        .Where(a => assignmentKinds.Contains(a.Kind()));
+                        .Where(a => AssignmentKinds.Contains(a.Kind()));
 
                     foreach (var assignment in assignments)
                     {
-                        ProcessExpression(assignment.Left);
+                        var identifierNames = IdentifierNames(assignment.Left);
```

Comment:
I suggest to use `leftSideExpressions` to make it clear from the name what we're looking at

and then the call below would be `ProcessExpression(leftSideExpression);`